### DISCUSSION TITLE
IsHeadersPresent() throws NullReferenceException in 5.2 and 6.8.1

### DISCUSSION THREAD

**Discussion by MartinBring** [rabbitmq/rabbitmq-dotnet-client#1632] [Q&A]
### Describe the bug

`Message.BasicProperties.IsHeadersPresent() `throws NullReferenceException sometimes (often), and sometimes not. It happens both in 5.2 and 6.8.1. It can have to do something how connections, channels, model are opened/closed. We experience that it doesn't happen with only one listener, but always after closing that one and starting two new connections/channels/listerners. Excuse me for not providing more information than this, i'm not that familiar with the area.

### Reproduction steps

**Working**
1. Open a connection
2. Start one channel with an and inputlistener to a queue
3. Put a message without headers on a queue.
2. Read from queue with Take()
4. Use Message.BasicProperties.IsHeadersPresent() to check for headers.
5. It will return false, no Exception occurs

**Not working**
1. Close the first connection
2. Start two new ones, each with a channel and inputlistener to different queues
3. Put a message without headers on one of the queues.
6. Read from queue with Take()
7. Use Message.BasicProperties.IsHeadersPresent() to check for headers.
8. NullReferenceException will occur

### Expected behavior

IsHeadersPresent() should return true or false, not throw any exception.

### Additional context

We now check the actual headers for null and Count > 0

`queueMessage.BasicProperties.Headers != null && queueMessage.BasicProperties.Headers.Count > 0`

**Comment by lechu445** [comment]
It may be worth to check if this issue still exists in v7.0.0-rc.5, as it might be fixed by the way of https://github.com/rabbitmq/rabbitmq-dotnet-client/issues/1596.

**Comment by lukebakken** [comment]
Hi @MartinBring -

`Take()` is not a method in this library. Could you provide source code that shows how you are reading messages from RabbitMQ?
